core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -222,7 +223,8 @@ public ThreadPool(Settings settings) {
         int halfProcMaxAt5 = Math.min(((availableProcessors + 1) / 2), 5);
         int halfProcMaxAt10 = Math.min(((availableProcessors + 1) / 2), 10);
         Map<String, Settings> defaultExecutorTypeSettings = new HashMap<>();
-        add(defaultExecutorTypeSettings, new ExecutorSettingsBuilder(Names.GENERIC).size(4 * availableProcessors).keepAlive("30s"));
+        int genericThreadPoolSize = Math.min(512, Math.max(128, 4 * availableProcessors));
```

Comment:
@bleskes I took @nik9000's suggestion and pushed 0337405b299a93d0225398cefd85e515b48574a9.